### ISSUE TITLE
Allow replacing images/files while retaining their custom fields' content

### ISSUE THREAD

**Issue by eelke** [processwire/processwire-requests#364]
### Short description of the enhancement

Allow replacing images/files while retaining their custom fields' content

### Optional: Steps that explain the enhancement

Quite simple:

1. Add media to an Image/File field with associated custom fields template (field-fieldname)
2. Add content to the custom fields
3. Receive improved content, for instance from an art department or client
4. Simply replace the image without having to re-enter all the content. 

Seems very logical, right?

### Current vs. suggested behavior

With the new [Custom Fields for images/files](https://processwire.com/blog/posts/pw-3.0.142/#custom-fields-for-files-images) it has become possible to add more data to an image than before, which is great. However, especially when using this feature in a media-rich site with a lot of custom fields, it quickly becomes apparent there is a problem: you cannot replace the image without destroying the custom fields' data as well.

I propose a drag/drop feature or upload button that allows the old image/file to be replaced, while retaining all the content of the custom fields that are associated with the old file. 

### Why would the enhancement be useful to users?

Just as an example, I'm sure there are plenty of other cases: We're in the process of building a very content-rich website, which is built around artworks that (of course) have a lot of meta-data. Dimensions, materials, owners, et cetera. This meta-data was entered by hand. But now the images need to be replaced with higher quality versions (official scans). While this should be a simple manual task, I had to write a special importer module in order to prevent the double work of having to re-enter everything.

**Comment by adrianbj** [comment]
This is absolutely essential for some sites - thanks for bringing it up.

I actually think it might even be nicer if there was a replace image/file button so that with fields that allow multiple images/files, it is more obvious what you are doing - ie replacing the file, but not the entry / metadata.

**Comment by matjazpotocnik** [comment]
For now you can replace the image directly on the filesystem, if you have access...

**Comment by eelke** [comment]
> For now you can replace the image directly on the filesystem, if you have access...

Indeed, that works and it is what my little importer script now does. By basically bypassing PW and replacing a file directly on the server. But it is cumbersome, and just not a good solution for my clients.

**Comment by gmclelland** [comment]
Just checking to see if you were aware of Files Rename Replace module?
https://processwire.com/talk/topic/19016-files-rename-replace/

I'm not sure if it would meet your needs or if the custom data get wiped as well?

In case you weren't aware, there are also some image replace techniques mentioned on https://processwire.com/blog/posts/major-images-field-upgrade/

Just thought I would mention it.

**Comment by Toutouwai** [comment]
Drag and drop replacement of images while keeping related data (description, tags) is an existing core feature: https://processwire.com/blog/posts/major-images-field-upgrade/

So the fact that custom images fields has broken this feature is a bug IMO, and there's an existing issue about it: https://github.com/processwire/processwire-issues/issues/1099
